plugins/autowarmth.koplugin/main.lua

```diff
@@ -0,0 +1,780 @@
+        table.sort(self.current_times, function(a,b) return a < b end)
+    end
+
+    if self.easy_mode then
+        self.current_times[1] = nil
+        self.current_times[2] = nil
+        self.current_times[3] = nil
+        self.current_times[6] = nil
+        self.current_times[9] = nil
+        self.current_times[10] = nil
+        self.current_times[11] = nil
+    end
+
+    -- here are dragons
+    local i = 1
+    -- find first valid entry
+    while not self.current_times[i] and i <= midnight_index do
+        i = i + 1
+    end
+    local next
+    while i <= midnight_index do
+        next = i + 1
+        -- find next valid entry
+        while not self.current_times[next] and next <= midnight_index do
+            next = next + 1
+        end
+        prepareSchedule(self.current_times, i, next)
+        i = next
+    end
+
+    local now = SunTime:getTimeInSec()
+
+    -- reschedule 5sec after midnight
+    UIManager:scheduleIn(24*3600 + 5 - now, self.scheduleMidnightUpdate, self )
+
+    self:scheduleWarmthChanges(now)
+end
+
+function AutoWarmth:scheduleWarmthChanges(time)
+    for i = 1, #self.sched_funcs do -- loop not essential, as unschedule unschedules all functions at once
+        if not UIManager:unschedule(self.sched_funcs[i][1]) then
+            break
+        end
+    end
+
+    local actual_warmth
+    for i = 1, #self.sched_funcs do
+        if self.sched_times[i] > time then
+            UIManager:scheduleIn( self.sched_times[i] - time,
+                self.sched_funcs[i][1], self.sched_funcs[i][2])
+        else
+            actual_warmth = self.sched_funcs[i][2] or actual_warmth
+        end
+    end
+    -- update current warmth directly
+    self.setWarmth(actual_warmth)
+end
+
+function AutoWarmth:hoursToClock(hours, withoutSeconds)
+    if hours then
+        hours = hours % 24 * 3600 + 0.01 -- round up, due to reduced precision in settings.reader.lua
+    end
+    return util.secondsToClock(hours, withoutSeconds)
+end
+
+function AutoWarmth:addToMainMenu(menu_items)
+    menu_items.autowarmth = {
+        text = _("Auto-warmth"),
+        checked_func = function() return self.activate ~= 0 end,
+        sub_item_table_func = function()
+            return self:getSubMenuItems()
+        end,
+    }
+end
+
+local function tidy_menu(menu, request)
+    for i = #menu, 1, -1 do
+        if menu[i].mode ~=nil then
+            if menu[i].mode ~= request then
+                table.remove(menu,i)
+            else
+                menu[i].mode = nil
+            end
+        end
+    end
+    return menu
+end
+
+local about_text = _([[Set the frontlight warmth and night mode based on a time schedule or the sun's position.
+
+There are three types of twilight:
+• Civil: You can read a newspaper.
+• Nautical: You can see the first stars.
+• Astronomical: It is really dark.
+
+Certain warmth-values can be set for every kind of twilight and sunrise, noon, sunset and midnight.
```

Comment:
Certain would imply that you can't fully customize it while being vague about what you can do, while I assume you can do whatever you want?
```suggestion
Custom warmth values can be set for every kind of twilight and sunrise, noon, sunset and midnight.
```